File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/dialog/bottomSheet/ImageSelectorBottomSheetDialogFragment.java
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.dialog.bottomSheet;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.BottomSheetDialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import butterknife.ButterKnife;
import butterknife.OnClick;
import de.karzek.diettracker.R;

/**
 * Created by MarjanaKarzek on 29.06.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 29.06.2018
 */
public class ImageSelectorBottomSheetDialogFragment extends BottomSheetDialogFragment {

    private OnOpenCameraClickListener cameraClickListener;
    private OnOpenGalleryClickListener galleryClickListener;

    public static ImageSelectorBottomSheetDialogFragment newInstance() {
        return new ImageSelectorBottomSheetDialogFragment();
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.snippet_bottom_sheet_image_selector, container,
                false);
        ButterKnife.bind(this, view);

        return view;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        try {
            cameraClickListener = (OnOpenCameraClickListener) getActivity();
            galleryClickListener = (OnOpenGalleryClickListener) getActivity();
        } catch (ClassCastException e) {
            throw new ClassCastException("fragment must implement OnOpenCameraClickListener and OnOpenGaleryClickListener");
        }
    }

    @OnClick(R.id.image_source_camera)
    void onOpenCameraClicked() {
        cameraClickListener.onOpenCameraClickedInBottomSheet();
        this.dismiss();
    }

    @OnClick(R.id.image_source_gallery)
    void onOpenGalleryClicked() {
        galleryClickListener.onOpenGalleryClickedInBottomSheet();
        this.dismiss();
    }

    public interface OnOpenCameraClickListener {
        void onOpenCameraClickedInBottomSheet();
    }

    public interface OnOpenGalleryClickListener {
        void onOpenGalleryClickedInBottomSheet();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        cameraClickListener = null;
        galleryClickListener = null;
    }
}

File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/dialog/bottomSheet/ImageSelectorBottomSheetDialogFragment.kt
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.dialog.bottomSheet

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import de.karzek.diettracker.R
import kotlinx.android.synthetic.main.snippet_bottom_sheet_image_selector.*

/**
 * Created by MarjanaKarzek on 29.06.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 29.06.2018
 */
class ImageSelectorBottomSheetDialogFragment : BottomSheetDialogFragment() {

    private var cameraClickListener: OnOpenCameraClickListener? = null
    private var galleryClickListener: OnOpenGalleryClickListener? = null

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.snippet_bottom_sheet_image_selector, container,
                false)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        try {
            cameraClickListener = activity as OnOpenCameraClickListener?
            galleryClickListener = activity as OnOpenGalleryClickListener?
        } catch (e: ClassCastException) {
            throw ClassCastException("fragment must implement OnOpenCameraClickListener and OnOpenGaleryClickListener")
        }

        image_source_camera.setOnClickListener {
            cameraClickListener!!.onOpenCameraClickedInBottomSheet()
            this.dismiss()
        }
        image_source_gallery.setOnClickListener {
            galleryClickListener!!.onOpenGalleryClickedInBottomSheet()
            this.dismiss()
        }
    }

    interface OnOpenCameraClickListener {
        fun onOpenCameraClickedInBottomSheet()
    }

    interface OnOpenGalleryClickListener {
        fun onOpenGalleryClickedInBottomSheet()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        cameraClickListener = null
        galleryClickListener = null
    }

    companion object {

        fun newInstance(): ImageSelectorBottomSheetDialogFragment {
            return ImageSelectorBottomSheetDialogFragment()
        }
    }
}
